src/reducecalculation.cpp

```diff
@@ -67,6 +67,31 @@ std::size_t GetReduceCalculationWorkspaceSize(Handle& handle,
 
         return pair_size_vector.empty() ? static_cast<size_t>(-1) : pair_size_vector.front().second;
     }
+    else if(reduceCalculationOp == MIOPEN_REDUCE_CALCULATION_ANY)
+    {
+        auto ctx           = ExecutionContext{&handle};
+        const auto problem = reduce::ProblemDescriptionCalculation{
+            MIOPEN_REDUCE_CALCULATION_NOT_PROPAGATE_NAN, xDesc, yDesc, dim, reduceCalculationOp};
+
+        const auto algo       = AlgorithmName{"AnyForward"};
+        const auto solvers    = solver::SolverContainer<solver::reduce::AnyForward>{};
+        auto pair_size_vector = solvers.GetWorkspaceSizes(ctx, problem);
+
+        return pair_size_vector.empty() ? static_cast<size_t>(-1) : pair_size_vector.front().second;
+    }
+    else if(reduceCalculationOp == MIOPEN_REDUCE_CALCULATION_ALL)
+    {
+        auto ctx           = ExecutionContext{&handle};
+        const auto problem = reduce::ProblemDescriptionCalculation{
+            MIOPEN_REDUCE_CALCULATION_NOT_PROPAGATE_NAN, xDesc, yDesc, dim, reduceCalculationOp};
```

Comment:
default for logicalreducecalculation should be "MIOPEN_REDUCE_CALCULATION_PROPAGATE_NAN"

Reply:
why not `const auto problem = reduce::ProblemDescriptionCalculation{
            nanPropagation, xDesc, yDesc, dim, reduceCalculationOp};` as `sum` and `prod`

Reply:
actually, for logical calculation, it doesnt use the value of nanPropagation, as the logical operations always treat `NaN` as true, hence there's no need to handle `NaN` like float types. Therefore, I think lets just keep `problem` variable initialization the same with sum and prod, which uses:
```
const auto problem = reduce::ProblemDescriptionCalculation{
            MIOPEN_REDUCE_CALCULATION_NOT_PROPAGATE_NAN, xDesc, yDesc, dim, reduceCalculationOp};
```